File: cletaeats_mobile/app/src/main/java/com/example/cletaeats_mobile/domain/interfaces/IRestauranteRepository.kt
package com.example.cletaeats_mobile.domain.interfaces

import com.example.cletaeats_mobile.domain.Result
import com.example.cletaeats_mobile.domain.model.Restaurante

interface IRestauranteRepository {
    suspend fun obtenerRestaurantes(): Result<List<Restaurante>>
}

File: cletaeats_backend/domain/interfaces/i_restaurante_repository.py
# domain/interfaces/i_restaurante_repository.py
from abc import ABC, abstractmethod
from typing import List, Optional
from core.entities.restaurante import Restaurante


class IRestauranteRepository(ABC):

    @abstractmethod
    def obtener_todos_activos(self) -> List[Restaurante]:
        pass

    @abstractmethod
    def obtener_por_id(self, id_restaurante: int) -> Optional[Restaurante]:
        pass